### ISSUE TITLE
Thumbor 4.12.2 - Firefox & Safari Issues

### ISSUE THREAD

**Issue by lcottingham** [thumbor/thumbor#939]
### Thumbor request URL
https://img.point-locker.club/ywRIkY99QaqtiNfy52R4b-Ibmzo=/fit-in/2560x1440/smart/https://cdn.databowlpages.com/assets/63cb46dc-adca-4f76-bd2e-67d0cb93b068.jpg

### Expected behaviour
Chrome (latest) - loads the image fine. If no cache available, processes, caches and feeds the completed image. Essentially, normal behaviour.

### Actual behaviour
Chrome - works fine.
Safari - Error and no load - 0kbs received
Firefox - Error and no load - 0kbs received

### Operating system
AWS EC2 - Linux AMI
Followed this guide exact: https://www.dadoune.com/blog/best-thumbnailing-solution-set-up-thumbor-on-aws/

### Your thumbor.conf
For testing - kept the thumbor.conf default.
Changes were to add SECRET_KEY and make unsafe requests and old url's FALSE.

I'm new to any form of server side setup - hence why I followed the guide to the tee. Checked logs for supervisord, thumbor.stderr and nginx. No issues.

Not looking for an exact fix, but i'm not sure what else to do and what tests to make now to resolve. Where do I go next to find out why Safari and Firefox don't like my installation? What is the community advice?

**Comment by masom** [comment]
@lcottingham I would suggest installing the latest available thumbor instead of locking to a specific version.

In the guide just `pip install thumbor` (without ==4.12.2) to get it.

`pip uninstall thumbor && pip install thumbor` would let you do that on an existing install.

**Comment by scorphus** [comment]
I confirm the link works fine on both browsers. Also, `curl` tells the correct content-type:

```
»»»» curl -v https://img.point-locker.club/ywRIkY99QaqtiNfy52R4b-Ibmzo=/fit-in/2560x1440/smart/https://cdn.databowlpages.com/assets/63cb46dc-adca-4f76-bd2e-67d0cb93b068.jpg > /dev/null
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 104.27.133.124...
* TCP_NODELAY set
* Connected to img.point-locker.club (104.27.133.124) port 443 (#0)
* TLS 1.2 connection using TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
* Server certificate: sni213930.cloudflaressl.com
* Server certificate: COMODO ECC Domain Validation Secure Server CA 2
* Server certificate: COMODO ECC Certification Authority
> GET /ywRIkY99QaqtiNfy52R4b-Ibmzo=/fit-in/2560x1440/smart/https://cdn.databowlpages.com/assets/63cb46dc-adca-4f76-bd2e-67d0cb93b068.jpg HTTP/1.1
> Host: img.point-locker.club
> User-Agent: curl/7.51.0
> Accept: */*
>
< HTTP/1.1 200 OK
< Date: Mon, 26 Jun 2017 18:07:57 GMT
< Content-Type: image/jpeg
< Content-Length: 470604
< Connection: keep-alive
< Set-Cookie: __cfduid=d39209e456bf80ebc56320da31674bd8f1498500477; expires=Tue, 26-Jun-18 18:07:57 GMT; path=/; domain=.point-locker.club; HttpOnly
< Expires: Mon, 10 Jul 2017 18:07:57 GMT
< Etag: "c3c85a963c556e1fb7cdffeab71195a71f3f9441"
< Cache-Control: public, max-age=1209600
< CF-Cache-Status: HIT
< Server: cloudflare-nginx
< CF-RAY: 37521bf2ceac50ce-MIA
<
{ [883 bytes data]
  3  459k    3 14573    0     0  23917      0  0:00:19 --:--:--  0:00:19 23890* Curl_http_done: called premature == 0
100  459k  100  459k    0     0   432k      0  0:00:01  0:00:01 --:--:--  433k
* Connection #0 to host img.point-locker.club left intact
```

**Comment by kkopachev** [comment]
Sometimes it returns webp image. Might be CDN misconfiguration
```
> GET /ywRIkY99QaqtiNfy52R4b-Ibmzo=/fit-in/2560x1440/smart/https://cdn.databowlpages.com/assets/63cb46dc-adca-4f76-bd2e-67d0cb93b068.jpg HTTP/1.1
> Host: img.point-locker.club
> User-Agent: curl/7.52.1
> Accept: */*
> 
{ [5 bytes data]
< HTTP/1.1 200 OK
< Date: Mon, 26 Jun 2017 19:52:25 GMT
< Content-Type: image/webp
< Content-Length: 288664
< Connection: keep-alive
< Set-Cookie: __cfduid=d655a546d83172b9fc19d3d109d604aad1498506745; expires=Tue, 26-Jun-18 19:52:25 GMT; path=/; domain=.point-locker.club; HttpOnly
< Expires: Mon, 10 Jul 2017 19:52:25 GMT
< Vary: Accept
< Etag: "f6b7019f7010c190f1f402bbd1d48880f2fe0e3a"
< Cache-Control: public, max-age=1209600
< CF-Cache-Status: HIT
< Server: cloudflare-nginx
< CF-RAY: 3752b4f4e8b851ac-SJC
< 
{ [869 bytes data]
* Curl_http_done: called premature == 0
```

**Comment by masom** [comment]
Oh, if auto_webp is enabled then the CDN might be storing it incorrectly (not using response headers) ?

**Comment by lcottingham** [comment]
Thanks for the responses!
I am looking at updating to the latest stable thumbor version but didn't want to change a pre-existing working install. I was going to spin up a duplicate and update a secondary instance... but I will :)

I did suspect it may be something to do with the webp image and it was activated. After posting this, I did change the config to turn webp FALSE and restarted my supervisord/nginx which may explain why it is now working.

I'll reset my browsers and take a look myself. For now, it's still not working for me in safari 10.1 and firefox (latest). I'll reset my cloudflare cache and try again. Slight relief that it is working for you all though.
